crates/sui-indexer-alt-graphql/src/api/types/epoch.rs

```diff
@@ -240,28 +257,28 @@ impl EpochStart {
 }
 
 impl EpochEnd {
-    fn empty() -> Self {
-        Self { contents: None }
-    }
-
     /// Attempt to fetch information about the end of an epoch from the store. May return an empty
     /// response if the epoch has not ended yet, as of the checkpoint being viewed.
-    async fn fetch(ctx: &Context<'_>, scope: &Scope, epoch_id: u64) -> Result<Self, RpcError> {
+    async fn fetch(
+        ctx: &Context<'_>,
+        epoch_start: &EpochStart,
+        epoch_id: u64,
+    ) -> Result<Self, RpcError> {
         let pg_loader: &Arc<DataLoader<PgReader>> = ctx.data()?;
-        let Some(stored) = pg_loader
+        let scope = epoch_start.scope.clone();
+        let stored = pg_loader
             .load_one(EpochEndKey(epoch_id))
             .await
-            .context("Failed to fetch epoch end information")?
-        else {
-            return Ok(Self::empty());
-        };
-
-        if stored.cp_hi as u64 > scope.checkpoint_viewed_at() {
-            return Ok(Self::empty());
-        }
+            .context("Failed to fetch epoch end information")?;
 
+        let end = match stored {
+            Some(end) if end.cp_hi as u64 >= scope.checkpoint_viewed_at() => Some(Arc::new(end)),
```

Comment:
I think this should be `<=`? i.e. we want to only return the `EpochEnd` if the epoch ended before `checkpoint_viewed_at`.
```suggestion
            Some(end) if end.cp_hi as u64 <= scope.checkpoint_viewed_at() => Some(Arc::new(end)),
```

Reply:
You are right, I accidentally switched this.